GameFiles/_integration.gsc

```diff
@@ -278,27 +297,71 @@ RegisterClientCommands()
 
 InitializeGameMethods() 
 {
-    level.overrideMethods = [];
-    level.overrideMethods["god"] = ::_god;
-    level.overrideMethods["noclip"] = ::UnsupportedFunc;
-    
-    if ( isDefined( ::God ) )
+    level.overrideMethods["UnsupportedFunc"]= ::UnsupportedFunc;
+    level notify("InitializeGameMethods");
+    level waittill("InitializeGameMethodsDone");
+}
+
+// Not every game can output to console or even game log.
+// Adds a very basic logging system that every
+// game specific script can extend.accumulate
+// Logging to dvars used as example.
+InitializeLogger()
+{
+    level.logger._logger = [];
+    RegisterLogger( ::Log2Dvar );
+    RegisterLogger( ::Log2IngamePrint );
+    level.logger.warning = ::LogWarning;
+    level.logger.error = ::LogError;
+}
+
+Log( LogLevel, message )
+{
+    for( i = 0; i < level.logger._logger.size; i++ )
     {
-        level.overrideMethods["god"] = ::God;
+        [[level.logger._logger[i]]]( LogLevel, message );
     }
-    
-    if ( isDefined( ::NoClip ) )
-    {
-        level.overrideMethods["noclip"] = ::NoClip;
+}
+
+LogWarning( message )
```

Comment:
might want to add a log debug here so we can reuse this method every place I do a check against whether debug is enabled or not.
eg: 
![image](https://user-images.githubusercontent.com/6732771/179421656-c9734439-9ddc-41bd-bb23-83f2c95940ba.png)

becomes `LogDebug( "...." )`